### PR TITLE
connection/jobs table update: track server connid, add user/id to jobs table

### PR DESCRIPTION
**Pull request by ixcat**
to fix issue #87, #275 (add user/server connection to jobs table).

ideally, connection ID would be directly available via MyM; for the moment,
hack in a query to retrieve connection ID on connection and use this for
jobs table logging.

supersedes: https://github.com/datajoint/datajoint-matlab/pull/276

addresses: https://github.com/datajoint/datajoint-matlab/pull/276#discussion_r506583746:
- [X] r.e. recurse: adjust to direct mym calls
- [X] r.e. indexing: feature too new, save ref and index that

status/remaining:
- [X] base functionality
- [X] unit test

### CODE REVIEW

**Comment by guzman-raphael** [comment]
@ixcat You will need to target a `stage-external-storage*` like branch first in order for the tests to work properly. Can you create another branch, e.g. `stage-external-storage2` since the current stage is already being used fro #303?

**Comment by ixcat** [comment]
ah right, got confused. will readjust.

**Review by dimitri-yatsenko — APPROVED** [review]

File: +dj/+internal/AutoPopulate.m [hunk 1]
```diff
@@ -149,8 +149,10 @@ function parpopulate(self, varargin)
             %   key=null          : blob                  # structure containing the key
             %   error_message=""  : varchar(1023)         # error message returned if failed
             %   error_stack=null  : blob                  # error stack if failed
+            %   user=""           : varchar(255)          # database user
             %   host=""           : varchar(255)          # system hostname
             %   pid=0             : int unsigned          # system process id
+            %   connection_id=0   : bigint unsigned       # system process id
```
Comment (guzman-raphael) [inline +dj/+internal/AutoPopulate.m:155]:
```suggestion
            %   connection_id=0   : bigint unsigned       # database connection id
```

File: +dj/+internal/AutoPopulate.m [hunk 2]
```diff
@@ -420,8 +424,10 @@ function createJobTable(self)
             fprintf(f, 'key=null           : blob                     # structure containing the key\n');
             fprintf(f, 'error_message=""   : varchar(1023)            # error message returned if failed\n');
             fprintf(f, 'error_stack=null   : blob                     # error stack if failed\n');
+            fprintf(f, 'user=""            : varchar(255)             # database user\n');            
             fprintf(f, 'host=""            : varchar(255)             # system hostname\n');
             fprintf(f, 'pid=0              : int unsigned             # system process id\n');
+            fprintf(f, 'connection_id=0    : bigint unsigned          # connection_id()\n');
```
Comment (guzman-raphael) [inline +dj/+internal/AutoPopulate.m:430]:
```suggestion
            fprintf(f, 'connection_id=0    : bigint unsigned          # database connection id\n');
```

File: +dj/Connection.m [hunk 3]
```diff
@@ -7,6 +7,7 @@
         use_tls
         inTransaction = false
         connId       % connection handle
+        serverId     % server connection ID
```
Comment (guzman-raphael) [inline +dj/Connection.m:10]:
```suggestion
        serverId     % database connection id
```

File: +dj/conn.m [hunk 4]
```diff
@@ -114,5 +114,7 @@
 end
 
 if nargout==0
-    query(connObj, 'SELECT connection_id()')
-end
\ No newline at end of file
+    fprintf('server connection id: %d\n', connObj.serverId);
```
Comment (guzman-raphael) [inline +dj/conn.m:117]:
```suggestion
    fprintf('database connection id: %d\n', connObj.serverId);    
```

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
Looking good! Thanks for getting this test case together; I like it :smile:.
Only a few minor nits on comment docs. Not necessary to implement exactly my suggestion but just looking to maintain some consistency.

File: +dj/+internal/AutoPopulate.m (hunk 1 above)
Comment (ixcat) [reply +dj/+internal/AutoPopulate.m]: good catch(es). updated / resolving all.

**Review by guzman-raphael — APPROVED** [review]

**Pull request merged**